src/refactorings/simplify-ternary/simplify-ternary.ts

```diff
@@ -0,0 +1,109 @@
+import { Editor, Code, ErrorReason } from "../../editor/editor";
+import { Selection } from "../../editor/selection";
+import * as t from "../../ast";
+
+export { simplifyTernary, hasTernaryToConvert };
+
+async function simplifyTernary(
+  code: Code,
+  selection: Selection,
+  editor: Editor
+) {
+  const updatedCode = updateCode(t.parse(code), selection);
+
+  if (!updatedCode.hasCodeChanged) {
+    editor.showError(ErrorReason.DidNotFoundTernaryToSimplify);
+    return;
+  }
+
+  await editor.write(updatedCode.code);
+}
+
+function hasTernaryToConvert(ast: t.AST, selection: Selection): boolean {
+  let result = false;
+
+  t.traverseAST(ast, {
```

Comment:
Could we re-use the `createVisitor()` to reduce the logic duplication? I think it would be easier to have only one series of conditions to maintain. In fact, `hasTernaryToConvert()` would emit `true` whenever `updateCode()` would replace a `path` with a node.

My suggestion to simplify would be:

1. Copy paste the logic of `updateCode()` (L54 to L92) to the `createVisitor()` callback function, instead of the current `onMatch(path)` call
2. Make `onMatch()` take an additional `t.Node` param (`onMatch(path, convertedNode)`)
3. Replace all `path.replaceWith(node)` with `onMatch(path, node)` call in the  copied logic
4. Replace the callback in `updateCode()` with `(path, convertedNode) => path.replaceWith(convertedNode)`
5. Replace the callback in `hasTernaryToConvert()` with `() => (result = true)`

What do you think?